processor/filterprocessor/metrics.go

```diff
@@ -168,6 +179,7 @@ func (fmp *filterMetricProcessor) processMetrics(ctx context.Context, md pmetric
 	if md.ResourceMetrics().Len() == 0 {
 		return md, processorhelper.ErrSkipProcessingData
 	}
+	fmp.telemetry.record(int64(totalMetrics-md.ResourceMetrics().Len()), int64(totalMetrics))
```

Comment:
This is recording the number of ResourcesMetrics dropped, but not the number of metrics themselves.  We'll need to move this recording somewhere else or update `totalMetrics` to count the number or metrics inside the `ResourceMetrics` slice.

Reply:
I believe I've fixed this, initially I did not understand the distinction between `ResourceMetrics`, `Metrics`, and `DataPoints`.